admin/admin_token.go

```diff
@@ -0,0 +1,64 @@
+package admin
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/vocdoni/blind-csp/types"
+	"go.vocdoni.io/dvote/crypto/ethereum"
+)
+
+/**
+ * GenerateAdminToken generates a new admin token for an election working for the current month
+ * Token is generated using HMAC-SHA256 with a secret key so that ensures that the token
+ * cannot be easily guessed or tampered with without knowledge of the key.
+ * Token will be valid for the "current" month (UTC) and the election ID.
+ */
+func GenerateAdminToken(electionId types.HexBytes) (string, error) {
+	secretKey := os.Getenv("CSP_ADMIN_TOKEN_SECRET")
+
+	currentTime := time.Now()
+	dateString := currentTime.Format("200601")
+
+	message := fmt.Sprintf("%s|%s", electionId.String(), dateString)
```

Comment:
This way of generating the token string is a bit weird... I'd propose the following:

1. Use go-dvote/crypto/ethereum `HashRaw([]byte) function` (performs a keccack256)
2. HashRaw( append([]byte(secretKey), append(electionId, []byte(dateString)...)...))